### PR TITLE
Crop width and height dimensions returned by hook

### LINKED ISSUE #22: Hotspots and Crops not respected
If  you have hotspots set or image is cropped using Sanity's built in tools, it's not respected with next-sanity-image.  There should be a way to either automatically or via a prop use any crop or hotspots set by the Sanity user.

example schema:

```
import { FaRegImage } from "react-icons/fa";
export default {
  name: "figure",
  title: "Image",
  type: "image",
  options: {
    hotspot: true,   // if this is set to true, user can set hotspots and crop the image, so their selected preference should follow to display
    metadata: ["lqip"],
  },
  icon: FaRegImage,
  fields: [
    {
      name: "alt",
      type: "string",
      title: "Alternative text",
      description: "Important for SEO and accessibility. Describe the image",
      options: {
        isHighlighted: true,
      },
      validation: (Rule) => [Rule.required().error("You need to set Alt text")],
    },
  ],
  preview: {
    select: {
      imageUrl: "asset.url",
      title: "alt",
    },
  },
};

```

### PR DESCRIPTION
**Pull request by gpoole**
Automatically adjusts the `width` and `height` dimensions returned from `useNextSanityImage` for images with a `crop` present.

I believe the hotspot component should be [automatically handled by @sanity/image-url](https://github.com/sanity-io/image-url#builder-methods) and it isn't necessary to touch it here.

I had to fix an unrelated failing test to get them working (using `DEFAULT_FALLBACK_IMAGE_WIDTH` which is not defined), I'm not sure if I missed something there.

Fixes #22

### CODE REVIEW

**Comment by RobEasthope** [comment]
@gpoole Thanks for the work on this.

@lorenzodejong This PR is a bit critical for my current project. Is the ready to go and if not, is there anything I can do to help such as testing the edge cases mentioned in the ticket?

**Comment by lorenzodejong** [comment]
Great work on the PR, i'll take a look at this with my test project. @RobEasthope if you want you can also test the scenario's in your project by checking out this PR locally.

File: src/useNextSanityImage.ts [hunk 1]
```diff
@@ -123,6 +142,7 @@ export function useNextSanityImage(
 		}
 
 		const originalImageDimensions = getImageDimensions(id);
+		const croppedImageDimensions = getCroppedDimensions(image);
 
 		const loader: ImageLoader = ({ width, quality }) => {
 			return (
```
Comment (lorenzodejong) [inline src/useNextSanityImage.ts:148]: Perhaps we should also pass the croppedImageDimensions to the imageBuilder callback, so it can be used to modify the image further.

File: src/useNextSanityImage.ts [hunk 2]
```diff
@@ -123,6 +142,7 @@ export function useNextSanityImage(
 		}
 
 		const originalImageDimensions = getImageDimensions(id);
+		const croppedImageDimensions = getCroppedDimensions(image);
```
Comment (lorenzodejong) [inline src/useNextSanityImage.ts:145]: It might be better to pass the `originalImageDimensions` to `getCroppedDimensions` to prevent calculating the original/base dimensions twice.

**Review by lorenzodejong — CHANGES_REQUESTED** [review]

File: src/useNextSanityImage.ts (hunk 1 above)
Comment (lorenzodejong) [reply src/useNextSanityImage.ts:148]: Make sure to also modify the `UseNextSanityImageBuilderOptions` and `UseNextSanityBlurUpImageBuilderOptions` types.

**Comment by RobEasthope** [comment]
> Great work on the PR, i'll take a look at this with my test project. @RobEasthope if you want you can also test the scenario's in your project by checking out this PR locally.

Will do @lorenzodejong

**Comment by lorenzodejong** [comment]
I've done some testing in my project and have not found an unsupported edge case. If we can fix the requested changes we can get this PR integrated.

**Comment by gpoole** [comment]
Thanks for the feedback, hopefully should all be addressed now @lorenzodejong.

**Review by lorenzodejong — APPROVED** [review]

**Comment by lorenzodejong** [comment]
Version 3.2.0 has just been released with this feature. Thanks for the contribution @gpoole, great work!

**Comment by lorenzodejong** [comment]
Version 3.2.0 included an incorrect build by my mistake, i've just released 3.2.1 which fixes this. I will implement a Github Action for releasing in the future to prevent these kinds of mistakes from happening...

**Comment by gpoole** [comment]
Thanks @lorenzodejong and thanks for this hook, saved me a lot of time!
